src/utils/auth/validate.request.body.ts

```diff
@@ -1,29 +1,31 @@
 import prismaClient from "../../client.js";
-
-export async function validateRequestBody(
-	username: string,
-	password: string,
-): Promise<boolean> {
+export async function validateRequestBody(username: string, password: string) {
 	if (!username || !password) return false;
-
 	// return an false if username or password contains invalid characters
 	if (
 		/[^a-zA-Z0-9_@$+]/gi.test(username) ||
 		/[^a-zA-Z0-9_@$+]/gi.test(password)
 	)
 		return false;
-
 	return true;
 }
-
 export async function isUsernameTaken(username: string) {
-	const userName = await prismaClient.user.findFirst({
-		where: {
-			username,
-		},
-	});
-
-	if (userName?.username === username) return true;
-
+	try {
+		const userName = await prismaClient.user.findFirst({
+			where: {
+				username,
+			},
+		});
+		if (
+			(userName === null || userName === void 0
+				? void 0
+				: userName.username) === username
+		)
+			return true;
```

Comment:
Remember when I told you about how white men don't  like code that gives them headache?, lol, this is a perfect example of one of them.

1. Why are you using tenary operators inside an if condition?. They literally do the same thing
2. Why are you using `void 0` instead of explicitly using `undefined`?. looks easier and more readable to me
3. Finally, the code is way too complex for just checking if a username exists lol. Why not just do 

```suggestion
return username ? true : false
```
or better still 
```suggestion
return !!username
```

Reply:
oversight i was goint to change it...